types/sharedb/lib/sharedb.d.ts

```diff
@@ -138,6 +138,19 @@ export class Doc<T = any> extends TypedEmitter<DocEventMap<T>> {
     flush(): void;
 }
 
+export type ConnectionState = 'connecting' | 'connected' | 'disconnected' | 'stopped' | 'closed';
+export type ConnectionStateEventMap = {
+    [state in ConnectionState]: (reason: string) => void;
+};
+export type ConnectionEventMap = ConnectionStateEventMap & {
+    'connection error': (error: Error) => void;
+    'doc': (doc: Doc) => void;
+    'error': (error: Error) => void;
+    'receive': (request: any) => void;
```

Comment:
Looks like this could be the narrower `{data: any}`
https://github.com/share/sharedb/blob/8b531bedf19860ffe0b37a3e1c8da7a32b5005bd/lib/client/connection.js#L139